### PR TITLE
FEATURE: ONE-4619 Add option to turn off datapoints

### PR DESCRIPTION
**Pull request by fiedlr**
(new PR, same story)

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by fiedlr** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/781/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2371/ : FAILURE in 20m 01s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/782/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/320/ : UNSTABLE in 25m 28s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by fiedlr** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/785/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2374/ : FAILURE in 17m 31s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/786/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/321/ : UNSTABLE in 22m 48s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/788/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2376/ : FAILURE in 11m 46s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by fiedlr** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/791/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/322/ : SUCCESS in 11m 47s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/790/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2378/ : FAILURE in 23m 26s

**Comment by lupko** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/794/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2383/ : FAILURE in 14m 27s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/795/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2384/ : FAILURE in 9m 19s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/799/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2387/ : FAILURE in 11m 08s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/800/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2388/ : SUCCESS in 9m 43s

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx [hunk 1]
```diff
@@ -60,4 +72,14 @@ export default class LineChartBasedConfigurationPanel extends BaseChartConfigura
...
+
+    protected isDataPointsControlDisabled(): boolean {
+        const properties = this.props.properties?.controls;
+
+        return (
+            this.props.type === "combo" &&
+            properties?.primaryChartType === "column" &&
+            properties?.secondaryChartType === "column"
+        );
+    }
```
Comment (lupko) [inline libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:84]: this looks suspicious. why should the LineBasedConfigPanel check for type 'combo' and types of primary/secondary charts?

indeed, the test must be done somewhere but a more sane place is in `PluggableComboChart` in `renderConfigurationPanel`. you can have this test there and then pass the result down to reused lineBasedConfig panel as a prop (dataPointsControlDisabled)

File: libs/sdk-backend-spi/src/common/settings.ts [hunk 2]
```diff
@@ -57,6 +57,11 @@ export enum SettingCatalog {
      * Indicates whether "Treat null as zero" option should be displayed in measure value filter configuration and its default state"
      */
     ADMeasureValueFilterNullAsZeroOption = "ADMeasureValueFilterNullAsZeroOption",
+
+    /**
+     * Indicates whether data point visibility configuration is available in AD
+     */
+    enableHidingOfDataPoints = "enableHidingOfDataPoints",
```
Comment (lupko) [inline libs/sdk-backend-spi/src/common/settings.ts:64]: all good. one little addition: please turn this option on by default for tiger. tiger's settings are at the moment hardcoded in the SDK (tiger does not have settings API yet). check out `sdk-backend-tiger`: `src/backend/workspace/settings` and `src/backenduser/settings.ts`

just add the key-value pair in there.

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx (hunk 1 above)
Comment (fiedlr) [reply libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:84]: You're right but the ConfigurationPanelContent class hierarchy makes it ugly to create custom props for the individual config panel types. I'll try to work out a solution though.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/807/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2395/ : SUCCESS in 14m 22s

**Review by lupko — APPROVED** [review]
LGTM

**Comment by lupko** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/336/ : FAILURE in 3m 40s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/592/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2397/ : SUCCESS in 14m 18s

**Pull request merged**